### PR TITLE
Genarate API and CLI documentation automatically.

### LINKED ISSUE #762: Add updating API documentation to build process
[lbry.io/api](https://lbry.io/api) is currently manually updated and regularly forgotten during the release process.

### PR DESCRIPTION
**Pull request by hackrush01**
@lyoshenka is there a way to automatically run these scripts when a PR is merged or during a version bump? If so then that would be great as then these would not have to be manually run.
Also short args(single dashed args) are now **unsupported**.
Also is there a place to setup a wiki article? I need to add rules on how the docstrings are to be made/maintained as the scripts do a bit of manual(hardcoded) parsing to be able to generate the docs.

@jackrobison you should probably look into the changes in `Daemon.py` file to see if there's something I missed. There are some CLI breaking changes there.

Closes #762

### CODE REVIEW

File: lbrynet/daemon/Daemon.py [hunk 1]
```diff
@@ -997,17 +997,16 @@ def _blob_availability(self, blob_hash, search_timeout, blob_timeout, downloader
     ############################################################################
 
     @defer.inlineCallbacks
-    @AuthJSONRPCServer.flags(session_status="-s", dht_status="-d")
```
Comment (jackrobison) [inline lbrynet/daemon/Daemon.py]: the `flags` decorator can be removed from `AuthJSONRPCServer`

File: lbrynet/daemon/Daemon.py [hunk 2]
```diff
@@ -1773,10 +1799,11 @@ def jsonrpc_stream_cost_estimate(self, uri, size=None):
         Get estimated cost for a lbry stream
 
         Usage:
-            stream_cost_estimate <uri> [<size> | --size=<size>]
+            stream_cost_estimate (<uri> | --uri=<uri>) [<size> | --size=<size>]
 
         Options:
-            <size>, --size=<size>  : stream size in bytes. if provided an sd blob won't be
+            --uri=<uri>      : (str) uri to use
+            --size=<size>  : (float) stream size in bytes. if provided an sd blob won't be
```
Comment (jackrobison) [inline lbrynet/daemon/Daemon.py]: please add (or remove) whitespace from these so the colons have matching indent levels (there are a few other instances)

File: lbrynet/daemon/Daemon.py [hunk 3]
```diff
@@ -1453,11 +1477,11 @@ def jsonrpc_claim_show(self, txid=None, nout=None, claim_id=None):
...
-            <txid>, --txid=<txid>              : look for claim with this txid, nout must
-                                                    also be specified
-            <nout>, --nout=<nout>              : look for claim with this nout, txid must
-                                                    also be specified
-            <claim_id>, --claim_id=<claim_id>  : look for claim with this claim id
+            --txid=<txid>              : (str) look for claim with this txid, nout must
+                                         also be specified
+            --nout=<nout>              : (int) look for claim with this nout, txid must
+                                         also be specified
+            --claim_id=<claim_id>  : (str) look for claim with this claim id
```
Comment (jackrobison) [inline lbrynet/daemon/Daemon.py]: here too

File: CHANGELOG.md [hunk 4]
```diff
@@ -28,6 +28,8 @@ at anytime.
   * daemon not erring properly for non-numeric values being passed to the `bid` parameter for the `publish` method
   * `publish` command to allow updating claims with a `bid` amount higher than the wallet balance, so long as the amount is less than the wallet balance plus the bid amount of the claim being updated (https://github.com/lbryio/lbry/issues/748)
   * incorrect `blob_num` for the stream terminator blob, which would result in creating invalid streams. Such invalid streams are detected on startup and are automatically removed (https://github.com/lbryio/lbry/issues/1124)
+  * fixed the inconsistencies in docstrings
```
Comment (jackrobison) [inline CHANGELOG.md]: inconsistencies in API docstrings

File: CHANGELOG.md [hunk 5]
```diff
@@ -88,7 +91,7 @@ at anytime.
   * unnecessary `TempBlobManager` class
   * old storage classes used by the file manager, wallet, and blob manager
   * old `.db` database files from the data directory
-  *
+  * short(single dashed) arguments
```
Comment (jackrobison) [inline CHANGELOG.md]: short (single dashed) arguments for `lbrynet-cli`

**Comment by jackrobison** [comment]
I haven't tested this yet, but looks great!

>There are some CLI breaking changes there.

Can you expand on this?

File: CHANGELOG.md (hunk 4 above)
Comment (hackrush01) [reply CHANGELOG.md]: Ok, but it is valid for both API and CLI
Will the following be better?
inconsistencies in API and CLI docstrings

File: CHANGELOG.md (hunk 5 above)
Comment (hackrush01) [reply CHANGELOG.md]: Ok

File: lbrynet/daemon/Daemon.py (hunk 1 above)
Comment (hackrush01) [reply lbrynet/daemon/Daemon.py]: From here?
https://github.com/lbryio/lbry/blob/master/lbrynet/daemon/auth/server.py#L153

File: lbrynet/daemon/Daemon.py (hunk 2 above)
Comment (hackrush01) [reply lbrynet/daemon/Daemon.py]: This will be automatically taken care by the scripts in the final output.

File: lbrynet/daemon/Daemon.py (hunk 3 above)
Comment (hackrush01) [reply lbrynet/daemon/Daemon.py]: This will be automatically taken care by the scripts in the final output.

**Comment by hackrush01** [comment]
CLI breaking changes are the removal of short flags.

Kind Regards
Amit Tulshyan

**Pull request merged**